src/portpicker.py

```diff
@@ -178,17 +178,20 @@ def _pick_unused_port_without_server():  # Protected. pylint: disable=invalid-na
             _random_ports.add(port)
             return port
 
-    # Try OS-assigned ports next.
+    # Next, try a few times to get an OS-assigned port.
     # Ambrose discovered that on the 2.6 kernel, calling Bind() on UDP socket
     # returns the same port over and over. So always try TCP first.
-    while True:
+    for _ in range(10):
```

Comment:
Picking 10 here is certainly open for debate. Figured we might as well try getting a port from the OS the same number of times we try a random port, since we have to check that it's free on UDP, too.